src/GraphQL/Resolvers/IFieldResolver.cs

```diff
@@ -1,14 +1,17 @@
+using GraphQL.Execution;
 using GraphQL.Types;
 
 namespace GraphQL.Resolvers
 {
     public interface IFieldResolver
     {
         object Resolve(ResolveFieldContext context);
+        object Resolve(ExecutionContext context, ExecutionNode node);
```

Comment:
> I don't think the ExecutionNode should be exposed in this way since it's internal to the execution

+1 to this, though also ditto with `ExecutionContext`.  That class is currently not exposed on the public API and I would prefer it to stay that way for now.  Is there a way we can accomplish something similar without exposing that class?

Reply:
True, actually the commit did not improve that much and i think i will revert and push the SimpleResolver interface in another PR.